Signal/src/ViewControllers/ConversationView/MessagesViewController.m

```diff
@@ -1176,19 +1175,12 @@ - (BOOL)showSafetyNumberConfirmationIfNecessaryWithConfirmationText:(NSString *)
                                                                        completion:completionHandler];
 }
 
-- (void)showFingerprintWithTheirIdentityKey:(NSData *)theirIdentityKey theirSignalId:(NSString *)theirSignalId
+- (void)showFingerprintWithRecipientId:(NSString *)recipientId
 {
     // Ensure keyboard isn't hiding the "safety numbers changed" interaction when we
     // return from FingerprintViewController.
     [self dismissKeyBoard];
 
-    OWSFingerprintBuilder *builder =
-        [[OWSFingerprintBuilder alloc] initWithStorageManager:self.storageManager contactsManager:self.contactsManager];
-    OWSFingerprint *fingerprint =
-        [builder fingerprintWithTheirSignalId:theirSignalId theirIdentityKey:theirIdentityKey];
-
-    NSString *contactName = [self.contactsManager displayNameForPhoneIdentifier:theirSignalId];
```

Comment:
I've DRY'd up this logic by pulling it into `FingerprintViewController`.  It also has the effect of ensuring that we're always showing the latest identity in that view.